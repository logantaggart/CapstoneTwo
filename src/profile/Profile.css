.account {
    text-align: center;
    border: seashell 5px solid;
    margin: 50px;
    color: seashell;
    padding: 50px;
}

.details {
    margin: 30px
}