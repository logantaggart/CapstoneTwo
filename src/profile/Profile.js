import React, { useContext } from 'react'
import UserContext from '../auth/UserContext'

import './Profile.css'

function Profile() {
    const { currentUser } = useContext(UserContext)

    return (
        <div className="account">
            <div className="details">
                <h2>Account Details:</h2>
                <hr />
                <br />
                <h5>
                    <b>Username:</b> {currentUser.username}
                </h5>
                <br />
                <h5>
                    <b>Full Name:</b> {currentUser.firstName} {currentUser.lastName}
                </h5>
                <br />
                <h5>
                    <b>Email:</b> {currentUser.email}
                </h5>
            </div>
        </div>
    )
}

export default Profile