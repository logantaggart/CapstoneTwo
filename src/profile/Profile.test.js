import React from 'react'
import { MemoryRouter } from 'react-router'

import { render } from '@testing-library/react'

import UserProvider from '../testUtils'

import Profile from './Profile'

it('renders without crashing', () => {
    render(
        <MemoryRouter>
            <UserProvider>
                <Profile />
            </UserProvider>
        </MemoryRouter>
    )
})

it('matches snapshot', () => {
    const { asFragment } = render(
        <UserProvider>
            <Profile />
        </UserProvider>
    )

    expect(asFragment()).toMatchSnapshot()
})