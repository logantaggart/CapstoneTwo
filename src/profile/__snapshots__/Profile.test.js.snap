// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="account"
  >
    <div
      class="details"
    >
      <h2>
        Account Details:
      </h2>
      <hr />
      <br />
      <h5>
        <b>
          Username:
        </b>
         testuser
      </h5>
      <br />
      <h5>
        <b>
          Full Name:
        </b>
          
      </h5>
      <br />
      <h5>
        <b>
          Email:
        </b>
         testuser@test.com
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
